monorepo_build.sh

```diff
@@ -28,16 +35,18 @@ for PARAM in $@; do
     if [ "$SUBDIRECTORY" == "" ]; then
         SUBDIRECTORY=$REMOTE
     fi
-    # Rewrite all branches from the first remote, only master branches from others
+
+    echo "Building all branches of the remote '$REMOTE'"
     if [ "$PARAM" == "$1" ]; then
-        echo "Building all branches of the remote '$REMOTE'"
-        $MONOREPO_SCRIPT_DIR/load_branches_from_remote.sh $REMOTE
+        $MONOREPO_SCRIPT_DIR/load_branches_from_remote.sh $REMOTE $SUBDIRECTORY --master
         $MONOREPO_SCRIPT_DIR/rewrite_history_into.sh $SUBDIRECTORY --branches
         MERGE_REFS='master'
     else
-        echo "Building branch 'master' of the remote '$REMOTE'"
+        $MONOREPO_SCRIPT_DIR/load_branches_from_remote.sh $REMOTE $SUBDIRECTORY
         git checkout --detach $REMOTE/master
-        $MONOREPO_SCRIPT_DIR/rewrite_history_into.sh $SUBDIRECTORY
+        # Re-write this current detached branch and also any branches starting
+        # with the $SUBDIRECTORY prefix
+        $MONOREPO_SCRIPT_DIR/rewrite_history_into.sh $SUBDIRECTORY "HEAD --branches=$SUBDIRECTORY--*"
```

Comment:
🍖 rewrite all branches checked out from this remote using the branch prefix

the final argument this script accepts is ultimately used as the argument to `git rev-list`, which is used by `git filter-branch` to rewrite the history of branches. Previously, there was no argument, and (I think) `HEAD` was the default value. Now, `git rev-list HEAD --branches=$SUBDIRECTORY--*` should match the current branch (detached remote master) _and_ all of this remote's imported branches, but no other remote's imported branches.

The re-writing is what moves the code into a subdirectory & updates indexes so that history is preserved.